content/post/elm-at-rakuten/index.md

```diff
@@ -347,193 +409,241 @@ This feature is so crucial that it inspired a new coding style: **Compiler Drive
 
 [^compile-driven-development]: Kevin Yank explains what is Compiler Driven Development [in this video](https://youtu.be/kuOCx0QeQ5c?t=1402). Louis Pilfold explains [how a compiler can be an assistant during software development](https://youtu.be/ceynSTa1dV4?t=987), referring to BEAM, a language inspired by Elm. And this is yet [another sample of Compiler Driven Development](https://youtu.be/jl1tGiUiTtI?t=1219) during one of the talks of Evan Czaplicki.
-
-[^type-driven-development]: [Idris](https://www.idris-lang.org/) is probably the fittest language for Type Drive Development. Idris has the feature of *[holes](http://docs.idris-lang.org/en/latest/elaboratorReflection/holes.html)*, while Elm can use *[Debug.todo](https://package.elm-lang.org/packages/elm/core/latest/Debug#todo)*.
+[^type-driven-development]: [Idris](https://www.idris-lang.org/) is probably the fittest language for Type Drive Development. Idris has the feature of _[holes](http://docs.idris-lang.org/en/latest/elaboratorReflection/holes.html)_, while Elm can use _[Debug.todo](https://package.elm-lang.org/packages/elm/core/latest/Debug#todo)_.
 
 > “If it compiles, it works” **[^if-it-compiles-it-works]**
 
 [^if-it-compiles-it-works]: This way of saying probably [originated in Haskell](https://wiki.haskell.org/Why_Haskell_just_works) but it applies to Elm too.
 
-Best practices should be automated as much as possible and the Elm compiler is playing an important role in this direction.
+Best practices should be automated as much as possible and the Elm compiler is playing an important role in this direction. **Elm makes “best practices” the default**.[^variable-shadowing]
+
+[^variable-shadowing]: An example of **Elm making “best practices” the default**, is about **variable shadowing** (same variable name defined twice in an ambiguous way). While most linters produce warnings in case of variable shadowing, the Elm compiler generates an error and stops the compilation, until the variable shadowing issue is removed. More on this in the [Variable Shadowing](https://github.com/elm/compiler/blob/master/hints/shadowing.md) article.
+
+The compiler guarantees that all edge cases are covered, something difficult to achieve with hand-made unit tests. Another advantage of the compiler static analysis is that it is extremely fast and can provide the exact location of errors.[^elm-review]
 
-The compiler guarantees that all edge cases are covered, something difficult to achieve with hand-made unit tests. Another advantage of the compiler static analysis is that it is extremely fast and can provide the exact location of errors.
+[^elm-review]: To push the static analysis further, other tools can be used, like [Elm-Review](https://package.elm-lang.org/packages/jfmengels/elm-review/latest/).
 
 ![John Carmack on Elm Errors](./images/john-carmack-on-errors.png)
```

Comment:
The Carmack tweet doesn't link to Twitter.

Reply:
Fixed